qiita_db/metadata_template/prep_template.py

```diff
@@ -75,6 +75,27 @@ class PrepTemplate(MetadataTemplate):
     _sample_cls = PrepSample
     _fp_id = convert_to_id("prep_template", "filepath_type")
     _filepath_table = 'prep_template_filepath'
+    _log_table = "prep_template_edit"
+
+    @staticmethod
+    def _update_analyses(pt_id):
```

Comment:
because delete is a class method, which means it needs to be a static method in order to update on delete. Is a good argument for moving the delete to be an instance method.